src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java

```diff
@@ -119,9 +120,26 @@ private boolean hasSunlight(Player p) {
         return (world.getTime() < 12300 || world.getTime() > 23850) && p.getEyeLocation().getBlock().getLightFromSky() == 15;
     }
 
-    private void checkForRadiation(Player p) {
+    private void checkForRadiation(PlayerProfile profile) {
+        HashedArmorpiece[] armor = profile.getArmor();
+        Player p = profile.getPlayer();
+
         // Check for a Hazmat Suit
-        if (!SlimefunUtils.isItemSimilar(p.getInventory().getHelmet(), SlimefunItems.SCUBA_HELMET, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getChestplate(), SlimefunItems.HAZMAT_CHESTPLATE, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getLeggings(), SlimefunItems.HAZMAT_LEGGINGS, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getBoots(), SlimefunItems.RUBBER_BOOTS, true)) {
+        boolean hasHazmat = false;
+        for (HashedArmorpiece armorPiece : armor) {
+
+            Optional<SlimefunArmorPiece> sfArmor = armorPiece.getItem();
+            if (!sfArmor.isPresent()) continue;
+
+            if (sfArmor.get().getID().equals("SCUBA_HELMET") ||
+                    sfArmor.get().getID().equals("HAZMAT_CHESTPLATE") ||
+                    sfArmor.get().getID().equals("HAZMAT_LEGGINGS") ||
+                    sfArmor.get().getID().equals("RUBBER_BOOTS")) {
+                hasHazmat = true;
```

Comment:
This should only be true if all items are hazmat gear. Right now, this is not the case.

Reply:
It never worked like that though. Only 1 piece always protected you from radiation. Idk if it's a bug but players could just do half slime armor and half hazmat suit for:
 + Water Breathing
 + Fire Resistance
 + Radiation Resistance
 + Jump Boost
 + Speed Boost

Even I use that set when I play.

Reply:
This is the radiation protection check though. And for that you ALWAYS needed the full set, as seen here, before you changed this behaviour:

https://github.com/TheBusyBiscuit/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java#L124

If any piece is missing, the radiation check continues.